alert-common/src/main/java/com/synopsys/integration/alert/common/workflow/processor/SummaryMessageContentProcessor.java

```diff
@@ -123,20 +120,41 @@ private ProviderMessageContent summarize(final ProviderMessageContent message) {
             }
         }
 
-        return collapseDuplicateComponentItems(summarizedCategoryItems);
+        return summarizedCategoryItems;
     }
 
-    // FIXME this seems to screw things up
     private Map<String, ComponentAttributeMap> collectComponentItemData(Set<ComponentItem> componentItemsForOperation) {
         Map<String, ComponentAttributeMap> itemsByCategory = new LinkedHashMap<>();
         for (final ComponentItem categoryItem : componentItemsForOperation) {
-            ComponentAttributeMap itemsOfSameName = itemsByCategory.computeIfAbsent(categoryItem.getComponentKeys().getShallowKey(), ignored -> new ComponentAttributeMap());
-            itemsOfSameName.putAll(categoryItem.getItemsOfSameName());
+            ComponentAttributeMap itemsOfSameName = categoryItem.getItemsOfSameName();
+            String summaryKey = createSummaryKey(categoryItem.getComponentKeys(), itemsOfSameName);
+
+            ComponentAttributeMap allItemsForSpecialKey = itemsByCategory.computeIfAbsent(summaryKey, ignored -> new ComponentAttributeMap());
+            allItemsForSpecialKey.putAll(itemsOfSameName);
         }
 
         return itemsByCategory;
     }
 
+    private String createSummaryKey(ComponentKeys componentKeys, ComponentAttributeMap itemsOfSameName) {
+        StringBuilder summaryKeyBuilder = new StringBuilder();
+        summaryKeyBuilder.append(componentKeys.getShallowKey());
+        for (List<LinkableItem> namedItems : itemsOfSameName.values()) {
+            if (namedItems.size() == 1) {
+                final LinkableItem item = namedItems
+                                              .stream()
+                                              .findAny()
+                                              .orElseThrow();
```

Comment:
What are your thoughts on the basic .orElseThrow() when we've already verified that there is exactly one item?